src/lib/parquet/src/Flow/Parquet/ThriftStream/TPhpFileStream.php

```diff
@@ -63,15 +62,16 @@ public function read($len)
 
     public function write($buf) : void
     {
-        while (TStringFuncFactory::create()->strlen($buf) > 0) {
+        while ($buf != '') {
```

Comment:
I just noticed, how terrible those default classes from Thrift 🤦 instead of using strlen, they are creating an object that is using strlen inside 🤦 🤦 🤦  I might need to look into other classes and create our own custom implementations just slightly optimized or maybe even optimize it directly in thrift repo

Reply:
Shouldn't this be strict check?

Reply:
ah shit 🤦 IDE replaced this for me and I didn't notice it's != instead of !==. It wont affect anything as there can't be anything different than string here, will correct that later. Do you know if there is a rule in phpstan/psalm maybe that could enforce strict checks?

Reply:
There is a cs-fixer rule: https://cs.symfony.com/doc/rules/strict/strict_comparison.html but is really dumb one, and will lead to replacing every `==` with `===` without checking the code, so objects comparison will be also replaced will fail after applying.